SU2_CFD/src/solvers/CScalarSolver.cpp

```diff
@@ -372,40 +373,26 @@ void CScalarSolver::CompleteImplicitIteration(CGeometry* geometry, CSolver** sol
   /*--- Update solution (system written in terms of increments) ---*/
 
   if (!adjoint) {
-    /*--- Update the turbulent solution. Only SST variants are clipped. ---*/
-
-    switch (config->GetKind_Turb_Model()) {
-      case SA:
-      case SA_E:
-      case SA_COMP:
-      case SA_E_COMP:
-      case SA_NEG:
 
-        SU2_OMP_FOR_STAT(omp_chunk_size)
-        for (unsigned long iPoint = 0; iPoint < nPointDomain; iPoint++) {
-          nodes->AddSolution(iPoint, 0, nodes->GetUnderRelaxation(iPoint) * LinSysSol[iPoint]);
-        }
-        END_SU2_OMP_FOR
-        break;
+    su2double density = 1.0;
+    su2double density_old = 1.0;
 
-      case SST:
-      case SST_SUST:
-
-        SU2_OMP_FOR_STAT(omp_chunk_size)
-        for (unsigned long iPoint = 0; iPoint < nPointDomain; iPoint++) {
-          su2double density = flowNodes->GetDensity(iPoint);
-          su2double density_old = density;
+    /*--- Update the turbulent solution. Only SST variants are clipped. ---*/
 
-          if (compressible) density_old = flowNodes->GetSolution_Old(iPoint, 0);
+    SU2_OMP_FOR_STAT(omp_chunk_size)
+    for (unsigned long iPoint = 0; iPoint < nPointDomain; iPoint++) {
+      /*--- Multiply the Solution var with density to get the conservative transported quantity, if necessary. ---*/
+      if (Conservative) {
+        density = flowNodes->GetDensity(iPoint);
+        density_old = compressible ? flowNodes->GetSolution_Old(iPoint, 0) : density;
+      }
```

Comment:
That will work I guess, but it throws away optimizations.
Better use if (conservative) (outside of the loop over variables) to call either AddConservativeSolution or AddClippedSolution.

Reply:
Ok I removed that but I just kept AddConservativeSolution renamed to AddClippedSolution and made the factor between Solution and Transported variable default to 1. Thus the difference between the if(Conservative) ...} else {..} is really just the density. See 817186f826785e43d5aa8ae397846b39698dabbd . Let me know what you think